farmerbot/internal/power.go

```diff
@@ -141,23 +168,78 @@ func calculateResourceUsage(nodes map[uint32]node) (uint64, uint64) {
 		totalResources.add(node.resources.total)
 	}
 
-	used := usedResources.cru + usedResources.hru + usedResources.mru + usedResources.sru
-	total := totalResources.cru + totalResources.hru + totalResources.mru + totalResources.sru
+	return usedResources, totalResources
+}
+
+func (f *FarmerBot) selectNodesToPowerOn(demand capacity) ([]node, error) {
+	var selectedNodes []node
+	remainingDemand := demand
+
+	for _, node := range f.nodes {
+		if node.powerState != off {
+			continue // Skip nodes that are already on or waking up
+		}
+
+		// Check if this node can contribute to the remaining demand
+		contribute := false
```

Comment:
do you think it is a good idea to find the least number of nodes to wakeup?
i am thinking of in some cases maybe node1 satisfy cru only and it will be added to selectedNodes to power on, then node2 will satisfy the rest of resources and it also can satisfy the initial demand of cru, so instead of powering on both, node2 will be enough.